File: docker/python/welcome.py
#!/usr/bin/env python3
"""
Welcome to your Python DevPod workspace! 🐍

This is a sample Python file to get you started.
You can run this file by:
1. Opening the terminal (Ctrl+`)
2. Running: python welcome.py
"""

import sys
import os
from datetime import datetime

def main():
    print("🎉 Welcome to your Python DevPod workspace!")
    print(f"🐍 Python version: {sys.version}")
    print(f"📁 Current directory: {os.getcwd()}")
    print(f"⏰ Current time: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    print()
    print("🚀 Quick start commands:")
    print("  python --version     # Check Python version")
    print("  pip list            # List installed packages")
    print("  pip install <package>  # Install new packages")
    print()
    print("📝 Try editing this file and running it again!")
    
    # Simple interactive example
    name = input("👋 What's your name? ")
    print(f"Nice to meet you, {name}! Happy coding! 🎯")

if __name__ == "__main__":
    main()

File: docker/nodejs/welcome.js
#!/usr/bin/env node
/**
 * Welcome to your Node.js DevPod workspace! ⚙️
 * 
 * This is a sample Node.js file to get you started.
 * You can run this file by:
 * 1. Opening the terminal (Ctrl+`)
 * 2. Running: node welcome.js
 */

const os = require('os');
const path = require('path');
const readline = require('readline');

console.log('🎉 Welcome to your Node.js DevPod workspace!');
console.log(`⚙️  Node.js version: ${process.version}`);
console.log(`📁 Current directory: ${process.cwd()}`);
console.log(`💻 Platform: ${os.platform()} ${os.arch()}`);
console.log(`⏰ Current time: ${new Date().toLocaleString()}`);
console.log();
console.log('🚀 Quick start commands:');
console.log('  node --version       # Check Node.js version');
console.log('  npm --version        # Check npm version');
console.log('  npm init -y          # Initialize a new project');
console.log('  npm install <package>  # Install packages');
console.log();
console.log('📝 Try editing this file and running it again!');
console.log();

// Simple interactive example
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('👋 What\'s your name? ', (name) => {
  console.log(`Nice to meet you, ${name}! Happy coding! 🎯`);
  
  // Show some Node.js capabilities
  console.log('\n🔧 Node.js Environment Info:');
  console.log(`- Memory usage: ${Math.round(process.memoryUsage().heapUsed / 1024 / 1024)} MB`);
  console.log(`- Uptime: ${Math.round(process.uptime())} seconds`);
  console.log(`- Working directory: ${process.cwd()}`);
  
  rl.close();
});